plutus-contract/src/Plutus/Contract/Wallet.hs

```diff
@@ -193,6 +196,11 @@ addOutputs :: PubKey -> Value -> Tx -> Tx
 addOutputs pk vl tx = tx & over Tx.outputs (pko :) where
     pko = Tx.pubKeyTxOut vl pk
 
+removeEmptyOutputs :: Tx -> Tx
+removeEmptyOutputs tx = tx & over Tx.outputs (filter (not . isEmpty)) where
+    isEmpty (Tx.TxOut{Tx.txOutValue, Tx.txOutDatumHash}) =
+        null (Value.flattenValue txOutValue) && isNothing txOutDatumHash
```

Comment:
Right, so this is not removing script outputs, which makes sense. Then I guess we're just going to get an error later that we can't create an output with nothing in it, but I'm not sure there's anything we can do about that.

Reply:
Worth a comment though.